networking_ccloud/tools/netbox_config_gen.py

```diff
@@ -213,13 +214,39 @@ def get_azs(self) -> List[conf.AvailabilityZone]:
             azs.append(conf.AvailabilityZone(name=site.slug, suffix=suffix, number=number))
         return sorted(azs, key=attrgetter('number'))
 
-    def get_cloud_vrfs(self) -> List[conf.VRF]:
+    def get_address_scopes(self, address_scope_vrf_maps_path: List[Path]) -> Dict[str, List[str]]:
+        result = defaultdict(list)
+        for p in address_scope_vrf_maps_path:
+            with p.open('r') as f:
+                asvm = yaml.safe_load(f)
+                for k, values in asvm.items():
+                    if not isinstance(k, str) or not k.endswith('address_scopes'):
+                        continue
+                    for i, item in enumerate(values):
+                        name = item.get('name')
+                        vrf = item.get('vrf')
+                        if not name:
+                            print(f'{p.as_posix()}, key {k}, item {i}, needs a "name" to be considered '
+                                  'a valid address-scope mapping')
+                            continue
+                        if not vrf:
+                            print(f'{p.as_posix()}, key {k}, item {i}, needs a "vrf" to be considered '
+                                  'a valid address-scope mapping')
+                            continue
+                        result[vrf.lower()].append(name)
```

Comment:
I tried building a shorter solution for this but it did end up not really being shorter. But for fun and giggles:
```python
for i, item in enumerate(values):
    valid = False
    for key in 'name', 'vrf':
        if not item.get(key):
            print(f'{p.as_posix()}, key {k}, item {i}, needs a "{key}" to be considered '
                  'a valid address-scope mapping')
            break
    else:
        valid = True
    if not valid:
        continue

    result[item['vrf'].lower()].append(item['name'])
```
So... let's keep yours I guess ;)

Reply:
I was actually thinking of something like that too, but figured since it is just 2 fields I would not benefit from it.